app/models/user.rb

```diff
@@ -9,6 +9,12 @@ class User < ActiveRecord::Base
   has_many :votes
   has_many :resource_tags, as: :taggable
   has_many :tags, through: :resource_tags
+  scope :with_statistics, ->{
+    joins("LEFT JOIN questions ON questions.user_id = users.id LEFT JOIN " \
```

Comment:
The lord Ikem, Arel was initially used for this, but I had to change to raw SQL because of a particular limitation, but I'm still trying to find a way around though.
```ruby
    def query
      user_table.project(user_table[Arel.star], question_table[:id].count.as("questions_asked"),
      answers_table[:id].count.as("answers_given")).outer_join(question_table).
      on(user_table[:id].eq(question_table[:user_id])).outer_join(answers_table).
      on(user_table[:id].eq(answers_table[:user_id])).group(user_table[:id]).to_sql
    end
```
but using `find_by_sql` doesn't return an `ActiveRecord::Relation/Collection` therefore it was discarding all the associated `includes` so something like `User.with_associations.find_by_sql(query)` discards the associations defined and just runs the `find_by_sql`.